compat-data-schema.md

```diff
@@ -0,0 +1,365 @@
+specific behavior, like if a property applies to a set of elements or another
+are sub-features.
+
+* For HTML, an element or an attribute are features, while specific values of an
+attribute are sub-features.
+
+* For Web APIs or JavaScript, an interface, a method, a property, a constructor
+are features, while arguments or special values of an enumerated type are
+sub-features.
+
+Each feature is identified by a unique hierarchy of strings. E.g the
+`text-align` property is identified by `css.properties.text-align`.
+
+The strings of an identifier are not intended to be displayed and are therefore
+not translatable.
+
+In JSON, this gives:
+
+    "data": {
+      "css": {
+        "properties": {
+          "text-align": {…},
+          …
+        },
+        …
+      },
+      …
+    }
+
+ That way, each feature is uniquely identifier, independently of the file it is
+ defined in.
+
+ The hierarchy of identifiers is not defined inside the schema. It is a
+ convention of the project using the schema.
+
+ In the MDN browser-compat case, it is:
+
+ _Note: this list will evolve as we migrate our data_
+
+     "data": {
+       "css": {
+         "properties": {…}
+         "pseudo-classes": {…}
+         "pseudo-elements": {…}
+         "at-rules": {…}
+       },
+       "html": {
+         "elements": {
+           …,
+           "<element-name>": {
+             …,
+             "<attribute-name>": {…},
+             …
+           }
+         }
+         "global_attributes": {…}  
+       }
+     }
+
+### Feature description
+
+A feature is represented by an identifier containing the `"__compat"` property.
+In this compat property, you'll find the list of sub-features associated to the
+feature.
+
+A feature has at least one sub-feature, representing the basic support. It is
+always named `"basic_support"`.
+
+The basic support feature represents the minimal set of functionality included
+when a feature is qualified of 'supported'. What this represents depends of the
+evolution of the feature over time, both in term of specification and of browser
+support. Another way of seeing it is to consider `"basic_support"` as representing all
+the functionality of the feature that doesn't have its own sub-feature(s).
+
+### Sub-feature
+
+A sub-feature is the basic entity having browser compatibility information. As
+explained in the previous paragraph, any feature has at least one sub-feature
+called `'basic_support'`, but it may many more.
+
+A sub-feature may have three properties.
+
+* __Sub-feature description__ contained in the `"desc"` property. It is a
+`string` that contains a human-readable description of the sub-feature. As it is
+intended to be used as a kind of caption or title for the feature, keep it short.
+the `<code>` and `<a>`, as well as the macros `{{cssxref}}`, `{{HTMLElement}}`,
+`{{htmlattrxref}}`, and `{{domxref}}` can be used. See the localization section
```

Comment:
I'm quite uncomfortable with including HTML in strings, as it suggests that people will then go inserting it into pages, hello XSS. I deliberately made the strings text-only in the original, for this reason. We should at least note that people using these strings must clean them before doing this. Or am I paranoid?

I'm more uncomfortable with including KS macros in strings, as it seems to tie the data to KS/MDN. If someone is writing an editor using the data, what are they supposed to do with these things?